lib/Dialect/Torch/Transforms/FuseQuantizedOps.cpp

```diff
@@ -175,16 +190,16 @@ class QuantizeOperandsPastCommutingOps : public OpRewritePattern<SrcOp> {
       // stack is empty, so oldOpd is now the corrected verion of the
       // SrcOp's original operand
       // convert operand -> SrcOp to oldOpd -> newMPTQTOp -> SrcOp
-      auto MPTQTOperands = dequantOpd.getDefiningOp()->getOperands();
+      auto MPTQTOperands = chain.dequantOpd.getDefiningOp()->getOperands();
       auto qTorchType =
-          cast<ValueTensorType>(dequantOpd.getType()).getOptionalDtype();
+          cast<ValueTensorType>(chain.dequantOpd.getType()).getOptionalDtype();
       auto newMPTQTType = rewriter.getType<ValueTensorType>(
           cast<ValueTensorType>(operands[i].getType()).getSizes(), qTorchType);
       operands[i] = rewriter.create<Aten_MakePerTensorQuantizedTensorOp>(
           loc, newMPTQTType, oldOpd, MPTQTOperands[1], MPTQTOperands[2]);
     }
 
-    if (!dequanted) {
+    if (operandChains.empty()) {
```

Comment:
I suppose this could get moved above the rewriting loop. 

Speaking of, how do you generally feel about returning a match failure if 
```cpp
operandChains.size() != QuantInfo<SrcOp>OperandsToQuantize.size()
```
?
This way, we won't get "half-quantized" ops, which generally cause issues lower in the pipeline. 
Pros of adding this match failure: more reliable compile
Cons of adding this match failure: less likely that people would notice their ops aren't getting QDQ'd properly.

Let me know what you think.

Reply:
I think this is a good idea. The problem with half-quantized ops is also that there is little chance of salvaging them in downstream passes, while leaving them in the QDQ form still allows for later handling in specialized patterns.

I also agree with the Cons, but I think currently users will have to make sure that their IR gets quantized sensibly anyways. As our implementation becomes more comprehensive, e.g. also covering fusion for per-channel quantization, we could throw an error if an op remains in QDQ form until the end. At this point the user still has the option to apply his own transformations or run on a backend which supports the float point operator.

I've implemented this suggestion and have added a warning message emitted if at least one but not all operands were successfully traced to a supported per-tensor dequant Op. Let me know if you think this is too verbose.

Reply:
I'm totally happy with the op warning as a compromise. Thanks for the changes.